app/src/main/java/openfoodfacts/github/scrachx/openfood/repositories/ProductRepository.java

```diff
@@ -846,7 +896,21 @@ private Boolean tableIsEmpty(AbstractDao dao) {
      */
     @Override
     public Single<InsightAnnotationResponse> annotateInsight(String insightId, int annotation) {
-        return robotoffApi.annotateInsight(insightId, annotation);
+        // if the user is logged in, send the auth, otherwise make it anonymous
+        final SharedPreferences userPref = OFFApplication.getInstance()
+            .getSharedPreferences("login", 0);
+
+        final String user = userPref.getString("user", "");
+        final String pass = userPref.getString("pass", "");
+
+        if (StringUtils.isNotBlank(user) && StringUtils.isNotBlank(pass)) {
+            final String baseAuth = "Basic " + Base64.encodeToString(
+                (user + ":" + pass).getBytes(), Base64.DEFAULT);
```

Comment:
Can we move this to utils? We might be needing this at other places too!

Reply:
mmh, not likely, it's a different mechanism for auth than regular login

Reply:
Doesn't look any different it's the standard Basic Auth. Honestly, we should use this with all our API endpoints (provided that the server supports it for all the APIs) that require authentication rather than sending passwords as plaintexts from the app.